src/gpu/d3d11/SDL_gpu_d3d11.c

```diff
@@ -1725,14 +1715,120 @@ static SDL_GpuGraphicsPipeline* D3D11_CreateGraphicsPipeline(
+                {
+                    pipeline->resourceLayout.resourceSets[i].resourceInfos[j].bindSlots[k].shaderStageFlag = SDL_GPU_SHADERSTAGE_VERTEX;
+                    pipeline->resourceLayout.resourceSets[i].resourceInfos[j].bindSlots[k].slot = vertStorageBufferCount;
+
+                    vertStorageBufferCount += 1;
+                }
+                else if (resourceType == SDL_GPU_RESOURCETYPE_UNIFORM_BUFFER)
+                {
+                    pipeline->resourceLayout.resourceSets[i].resourceInfos[j].bindSlots[k].shaderStageFlag = SDL_GPU_SHADERSTAGE_VERTEX;
+                    pipeline->resourceLayout.resourceSets[i].resourceInfos[j].bindSlots[k].slot = vertUniformBufferCount;
+
+                    vertUniformBufferCount += 1;
+                }
+                else
+                {
+                    SDL_LogError(SDL_LOG_CATEGORY_APPLICATION, "Unsupported resource type!");
+                    return NULL;
+                }
+
+                k += 1;
+            }
+            else if (shaderStageFlags & SDL_GPU_SHADERSTAGE_FRAGMENT)
+            {
+                if (resourceType == SDL_GPU_RESOURCETYPE_TEXTURE_SAMPLER)
+                {
+                    pipeline->resourceLayout.resourceSets[i].resourceInfos[j].bindSlots[k].shaderStageFlag = SDL_GPU_SHADERSTAGE_FRAGMENT;
+                    pipeline->resourceLayout.resourceSets[i].resourceInfos[j].bindSlots[k].slot = fragSamplerCount;
+
+                    fragSamplerCount += 1;
+                }
+                else if (resourceType == SDL_GPU_RESOURCETYPE_STORAGE_TEXTURE_READONLY)
+                {
+                    pipeline->resourceLayout.resourceSets[i].resourceInfos[j].bindSlots[k].shaderStageFlag = SDL_GPU_SHADERSTAGE_FRAGMENT;
+                    pipeline->resourceLayout.resourceSets[i].resourceInfos[j].bindSlots[k].slot = fragStorageTextureCount;
+
+                    fragStorageTextureCount += 1;
+                }
+                else if (resourceType == SDL_GPU_RESOURCETYPE_STORAGE_BUFFER_READONLY)
+                {
+                    pipeline->resourceLayout.resourceSets[i].resourceInfos[j].bindSlots[k].shaderStageFlag = SDL_GPU_SHADERSTAGE_FRAGMENT;
+                    pipeline->resourceLayout.resourceSets[i].resourceInfos[j].bindSlots[k].slot = fragStorageBufferCount;
+
+                    fragStorageBufferCount += 1;
+                }
+                else if (resourceType == SDL_GPU_RESOURCETYPE_UNIFORM_BUFFER)
+                {
+                    pipeline->resourceLayout.resourceSets[i].resourceInfos[j].bindSlots[k].shaderStageFlag = SDL_GPU_SHADERSTAGE_FRAGMENT;
+                    pipeline->resourceLayout.resourceSets[i].resourceInfos[j].bindSlots[k].slot = fragUniformBufferCount;
+
+                    fragUniformBufferCount += 1;
+                }
+                else
+                {
+                    SDL_LogError(SDL_LOG_CATEGORY_APPLICATION, "Unsupported resource type!");
```

Comment:
i'm guessing it's intended that rw storage textures/buffers aren't supported here?

Reply:
binding UAVs in the graphics stage on D3D11 is cursed